mhkit/loads.py

```diff
@@ -79,8 +80,134 @@ def bin_statistics(data,bin_against,bin_edges,data_signal=[]):
+
+    # calculate TSR
+    TSR = rotor_velocity / inflow_speed
+
+    return TSR
+
+def calculate_Cp(power,inflow_speed,capture_area,rho):
+    '''
+    Function that calculates the power coefficient of device
+
+    Parameters
+    -------------
+    power : numpy array
+        Power output signal of device after losses [kW]
+    inflow_speed : numpy array
+        Speed of inflow [m/s]
+    capture_area : float/int
+        Projected area of rotor normal to inflow [m^2]
+    rho : float/int
+        Density of environment [kg/m^3]
+
+    Returns
+    -------------
+    Cp : numpy array
+        Power coefficient of device [-]
+
+    '''
+    # check data types
+    try:
+        power = np.asarray(power)
+        inflow_speed = np.asarray(inflow_speed)
+    except:
+        pass
+
+    assert isinstance(power, np.ndarray), 'power must be of type np.ndarray'
+    assert isinstance(inflow_speed, np.ndarray), 'inflow_speed must be of type np.ndarray'
+    assert isinstance(capture_area, (float,int)), 'capture_area must be of type int or float'
+    assert isinstance(rho, (float,int)), 'rho must be of type int or float'
+
+    # calculat power in
+    P_in = (0.5 * rho * capture_area * inflow_speed**3)/1000
+
+    # calculate Cp ratio
+    Cp = power / P_in 
+
+    return Cp
+
+def calculate_blade_moments(blade_matrix,flap_offset,flap_raw,edge_offset,edge_raw):
+    '''
+    Transfer function for deriving blade flap and edge moments using blade matrix.
+
+    Parameters
+    -------------
+    blade_matrix : numpy array
+        Derived blade calibration coefficients [D1,D2,D3,D4]
+    flap_offset : float/int
+        Derived offset of raw flap signal obtained during calibration process
+    flap_raw : numpy array
+        Raw strain signal of blade in the flapwise direction
+    edge_offset : float/int
+        Derived offset of raw edge signal obtained during calibration process
+    edge_raw : numpy array
+        Raw strain signal of blade in the edgewise direction
+    
+    Returns:
+    -------------
+    M_flap : numpy array
+        Blade flapwise moment in engineering units
+    M_edge : numpy array
+        Blade edgewise moment in engineering units
+    '''
+    # check data types
+    try:
+        blade_matrix = np.asarray(blade_matrix)
+        flap_raw = np.asarray(flap_raw)
+        edge_raw = np.asarray(edge_raw)
+    except:
+        pass
+            
+    assert isinstance(blade_matrix, np.ndarray), 'blade_matrix must be of type np.ndarray'
+    assert isinstance(flap_offset, (float,int)), 'flap_offset must be of type int or float'
+    assert isinstance(flap_raw, np.ndarray), 'flap_raw must be of type np.ndarray'
+    assert isinstance(edge_offset, (float,int)), 'edge_offset must be of type int or float'
+    assert isinstance(edge_raw, np.ndarray), 'edge_raw must be of type np.ndarray'
+
+    # remove offset from raw signal
+    flap_signal = flap_raw - flap_offset
+    edge_signal = edge_raw - edge_offset
```

Comment:
Yes I am open to simplifying this. My only fear would be for someone to forget to apply it which could skew results without it being obvious. But this may be overcome by making it very apparent in the docstring that the input must already be corrected. @rpauly18 any thoughts?

Reply:
I think we should leave the offsets as input because I have the same fears as Chris. Users will forget to apply the offsets. In my experience users do not read the docs that thoroughly, or even if they do, over time they may forget that they need to apply the offsets themselves. Adding a couple numbers to the function call will not be a heavy lift for the user.